src/lib/services/api-token-service.ts

```diff
@@ -185,10 +188,19 @@ export class ApiTokenService {
             );
         }
 
+        const nextAllowedQuery = this.queryAfter.get(secret) ?? 0;
         if (
             !token &&
+            isPast(nextAllowedQuery) &&
             this.flagResolver.isEnabled('queryMissingTokens', flagContext)
         ) {
+            if (this.queryAfter.size > 1000) {
+                // establish a max limit to query after size to prevent memory leak
```

Comment:
I think it's more readable this way
```suggestion
                // establish a max limit for queryAfter size to prevent memory leak
```